population.cpp

```diff
@@ -203,13 +206,15 @@ std::vector<double> rescale_dist_to_fit(std::vector<double> distance_from_target
     return fitness_inds;
 }
 
-void reproduce(population& p, std::mt19937_64& rng)
+population reproduce(population p, std::mt19937_64& rng)
```

Comment:
This for example will not make the programme work unfortunately. When you get a population by value, at least from my debugging experience in the last few days, causes problems with the templating cause the population p here is a copy of the real poulation, and therefore the program first constructs a default population and then fills it with the info of the original one. But unfortunately for now it cannot seem to notice the type of network, and just builds it with plain network, that therefore do not mutate.

Reply:
Also, this is also not really a getter function.